mppi_path_planning_node:
  ros__parameters:
    mppi_path_planning:
      tf_frame:
        map: "map"
        base_link: "base_link"
      topic_name:
        map: "/lrf/storage_map"
        target: "/goal_pose"
        odom: "/odom"
        cmd_vel: "/auto/cmd_vel"
      planning_period: 0.01
      min_vel: [-0.3, 0.0, -1.0]
      max_vel: [0.3, 0.0, 1.0]
      goal:
        pos_range: 0.4 #[m]
        angle_range: 6.0 #[deg]
        min_vel_range: 0.5 #[m/s]
        min_angular_range: 0.5 #[rad/s]
      mppi:
        T: 40
        K: 1000
        dt: 0.1 # [s]
        lambda: 1.0 # temperature parameter
        alpha: 0.85
        sigma: [0.1, 0.0001, 0.25]
        weight:
          state: [0.0, 0.0, 0.0, 800.0, 800.0, 0.0]
          terminal: [0.0, 0.0, 0.0, 1200.0, 1200.0, 1.0]
          control: [12.0, 0.0, 5.0]
        obstacle_cost: 2000.0
        window_size: 20.0 # 70.0
